### PR TITLE
[Chore] Modernize project setup

### PR DESCRIPTION
**Pull request by kittinunf**
## Description

This PR aims to modernize the project gradle usage.

- [x] Remove all of `compile` and use `implementation` counterparts
- [x] Migrate Gradle to `5.2` instead of `4.10`
- [x] Fix inconsistencies around the projects (move files, take out JSON test from Android module etc.)

## Type of change

Check all that apply

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (a change which changes the current internal or external interface)
- [ ] This change requires a documentation update

## How Has This Been Tested?

In case you did not include tests describe why you and how you have verified the
changes, with instructions so we can reproduce. If you have added comprehensive
tests for your changes, you may omit this section.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if necessary
- [ ] My changes generate no new compiler warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

**Comment by kittinunf** [comment]
Please ignore CircleCI build for now.

File: gradlew [hunk 1]
```diff
@@ -28,7 +28,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m"'
```
Comment (NikkyAI) [inline gradlew]: i think this can be set in gradle.properties
and not be wiped on wrapper upgrade

Comment (kittinunf) [reply gradlew]: it is updated automatically by using the wrapper update ...

**Comment by kittinunf** [comment]
@NikkyAI Agree I will use `api` for `fuel-coroutine`

**Comment by SleeplessByte** [comment]
Apart from Nikkys comments, lgtm

File: gradlew (hunk 1 above)
Comment (kittinunf) [reply gradlew]: revert!

**Comment by kittinunf** [comment]
It looks like when updated to Gradle 5.2, it will break the test 🤔 🤦‍♂️

**Comment by iNoles** [comment]
"To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/5.2/userguide/gradle_daemon.html."

**Comment by kittinunf** [comment]
@iNoles I thought we set it here `org.gradle.daemon=true`, daemon should be enabled by default no?

File: sample/build.gradle.kts [hunk 2]
```diff
@@ -8,20 +7,24 @@ plugins {
 }
 
 dependencies {
-    implementation(Kotlin.stdlib)
+    // fuel related libraries
```
Comment (iNoles) [inline sample/build.gradle.kts]: these should be implementation because api should be only in the libraries.

Comment (kittinunf) [reply sample/build.gradle.kts]: it is a sample app, so ... 🤷‍♂️ ww

Comment (kittinunf) [reply sample/build.gradle.kts]: Addressed!

**Pull request merged**